In [16]:
from astroquery.gaia import Gaia

In [18]:
query = """
SELECT
    gaia.source_id, gaia.ra, gaia.dec, gaia.phot_g_mean_mag, gaia.bp_rp, gaia.parallax,
    tmass.j_m, tmass.h_m, tmass.ks_m, tmass.ph_qual
FROM gaiadr3.gaia_source AS gaia
JOIN gaiadr3.tmass_psc_xsc_best_neighbour AS xmatch
    ON gaia.source_id = xmatch.source_id
JOIN gaiadr1.tmass_original_valid AS tmass
    ON xmatch.original_ext_source_id = tmass.designation
WHERE
    DISTANCE(POINT('ICRS', gaia.ra, gaia.dec), POINT('ICRS', 132.825, 11.8)) < 1
    AND gaia.phot_g_mean_mag < 14
"""

job = Gaia.launch_job_async(query)
results = job.get_results()
print(len(results))


INFO: Query finished. [astroquery.utils.tap.core]
1018


In Text cell above is the query to search gaiadr3 - the distance point is based off of the M67 cluster as requested in the assignment, and the mean magnitude less than 14
The number of stars returned is 1018.

In [20]:
cuts = (results['ph_qual'] == 'AAA') & (results['parallax'] > 0)
cut_results = results[cuts]
print(len(cut_results))

988


The cell above cuts all ph_qual that does not satisfy the re